ONSdigital/dp-api-clients-go
# Feature/cantabular population types

### What

Added `GetPopulationTypes` method to query cantabular for a list of all dataset names as part of adding the `/census` endpoint to `dp-dataset-api` (WIP: https://github.com/ONSdigital/dp-dataset-api/pull/352)

Upgraded the version of Convey to support the `SoMsg` keyword

### How to review

The easy way:
1. Check the graph QL query implied by the structure of `PopulationTypeQuery` (in `population_types.go`) matches the desired query `query { datasets { name } }`
2. Check the test coverage of this new method

The hard way:
Write a test harness to integration test this new method against cantabular _or_ pull the `feature/create-census-root-endpoint` of `dp-datset-api` and check that the `/census` endpoint is working

### Who can review

Any gopher
